core/src/main/java/io/questdb/cairo/sql/SqlExecutionCircuitBreakerWrapper.java

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.sql;
+
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Misc;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+// does not have to be thread-safe, because worker threads own the jobs/contexts
+// which hold SqlExecutionCircuitBreakerWrapper objects
+public class SqlExecutionCircuitBreakerWrapper implements SqlExecutionCircuitBreaker, Closeable {
+    private SqlExecutionCircuitBreaker delegate;
+    private NetworkSqlExecutionCircuitBreaker networkSqlExecutionCircuitBreaker;
+
+    @Override
+    public void cancel() {
+        delegate.cancel();
+    }
+
+    @Override
+    public boolean checkIfTripped(long millis, long fd) {
+        return delegate.checkIfTripped(millis, fd);
+    }
+
+    @Override
+    public boolean checkIfTripped() {
+        return delegate.checkIfTripped();
+    }
+
+    @Override
+    public void close() throws IOException {
+        delegate = Misc.freeIfCloseable(networkSqlExecutionCircuitBreaker);
```

Comment:
nit: `NetworkSqlExecutionCircuitBreaker` is closeable
```suggestion
        delegate = Misc.free(networkSqlExecutionCircuitBreaker);
```